### ISSUE TITLE
pid1: handle mounts backed by volume managers more gracefully

### ISSUE THREAD

**Issue by YHNdnzj** [systemd/systemd#29591]
### Component

systemd

### Is your feature request related to a problem? Please describe

For mount units, we currently set `BindsTo=` or `StopPropagatedFrom=` dependencies on the backing device by default. This is not suitable for mounts backed by volume managers though, since in that case it's OK for the original backing device to disappear. We should handle things like device mapper renaming more gracefully.

### Describe alternatives you've considered

The said dependencies can be disabled through `x-systemd.device-bound=no` (https://github.com/systemd/systemd/pull/29549). But that requires the user to manually track & modify every mount entry involved, which is burdensome.

### The systemd version you checked that didn't have the feature you are asking for

latest main

**Comment by YHNdnzj** [comment]
See also: #16801 #29543

cc @yuwata

**Comment by YHNdnzj** [comment]
More related issues: https://github.com/systemd/systemd/issues/14674 https://github.com/systemd/systemd/issues/19393

**Comment by yuwata** [comment]
Also related to #8234.

**Comment by poettering** [comment]
> since in that case it's OK for the original backing device to disappear.

Hu? what? why is the backing device allowed to disappear?

**Comment by YHNdnzj** [comment]
> > since in that case it's OK for the original backing device to disappear.
> 
> Hu? what? why is the backing device allowed to disappear?

On a single-device btrfs, you can use `btrfs replace` to replace the backing device completely, and that can be done online.

I guess LVM can do something similar too, but not familiar with that.

**Comment by poettering** [comment]
the btrfs/bcachefs case and the LVM case is systematically different btw. btrfs/bcachefs are not actually block based, i.e. stat() on them returns an st_dev that points to a virtual device, i.e. one with major=0. This is different for ext4/xfs/vfat and the log, which point back to the actual backing device. (let's call these "pseudo-block-based")

btrfs/bcachefs do not really agree on how to denote backing devices, we simply don't know how to wait for them at all really. if you only have one backing device it might work, but otherwise we are just fucked.

I think at this point we should probably imply x-systemd.device-bound=no for btrfs/bcachefs, since they are only pseudo-block-based, and hence it's fine if any of the backing devices disappear, given they might have been replaced dynamically.

**Comment by poettering** [comment]
anyway, please don't call btrfs/bcachefs "volume managers". They work very very differently from LVM. LVM provides block devices, they should never disappear while ext4/xfs are mounted off them.

**Comment by YHNdnzj** [comment]
Yeah, maybe they operate differently under the hood, but the idea/original issue is that the name/devpath of the backing device could change. `lvrename` would result in us unmounting the fs too, and from that perspective btrfs replace is not very different.
